packages/subcommands/src/lib/Subcommand.ts

```diff
@@ -15,10 +15,43 @@ import {
 
 export class Subcommand<PreParseReturn extends Args = Args, O extends Subcommand.Options = Subcommand.Options> extends Command<PreParseReturn, O> {
 	public parsedSubcommandMappings: SubcommandMappingArray;
+	private caseInsensitiveSubCommands = false;
 
 	public constructor(context: PieceContext, options: O) {
 		super(context, options);
 		this.parsedSubcommandMappings = options.subcommands ?? [];
+
+		const clientOptions = this.container.client.options;
+		if (clientOptions.caseInsensitiveCommands) {
+			this.caseInsensitiveSubCommands = true;
+
+			// Because slash commands must be lowercase anyway, we can transform all to lowercase.
+			for (const cmd of this.parsedSubcommandMappings) {
+				cmd.name = cmd.name.toLowerCase();
+			}
+		}
+
+		if (options.generateDashLessAliases) {
+			for (const mapping of this.parsedSubcommandMappings) {
+				if (!Reflect.has(mapping, 'messageRun') || (Reflect.has(mapping, 'messageRun') && Reflect.has(mapping, 'chatInputRun'))) {
```

Comment:
Why does this skip if both handlers are specified?

Reply:
If we have a dash within the chat input command and we modify that here, it would be pretty confusing- the user would also have to register that comamand (or let sapphire do it for us)

Reply:
Whatever we go for, it won't be perfect. But I think the safest way to go about it is to only apply dashless when 
1. The option is enabled
2. The command *only* has `messageRun`
3. The command does NOT have `chatInputRun`

Reply:
Adding to that, I'd rather have it affect only message based commands because slash commands also deal with the Discord API in a sense (even though the user has to register those themselves) whereas message don't have that.

Reply:
So as I said, this should not care about the chatInputRun mapping at all, and it should add all messageRun mappings